app/workers/education_form/process_10203_submissions.rb

```diff
@@ -0,0 +1,164 @@
+      process_user_submissions(user_submissions)
+    end
+
+    private
+
+    # Group the submissions by user_uuid
+    def group_user_uuid(records)
+      records.group_by { |ebc| ebc.education_stem_automated_decision&.user_uuid }
+    end
+
+    # If the user doesn't have EVSS data mark the 10203 as DENIED
+    # If there are multiple submissions for a user compare un-submitted to most recent processed
+    #   by EducationForm::CreateDailySpoolFiles
+    # Otherwise check submission data and EVSS data to see if submission can be marked as PROCESSED
+    def process_user_submissions(user_submissions)
+      user_submissions.each do |user_uuid, submissions|
+        gi_bill_status = get_gi_bill_status(user_uuid)
+        if gi_bill_status == {} || gi_bill_status.remaining_entitlement.blank?
+          submissions.each { |submission| update_status(submission, DENIED) }
+        elsif submissions.count > 1
+          check_previous_submissions(submissions, gi_bill_status)
+        else
+          process_submission(submissions.first, gi_bill_status)
+        end
+      end
+    end
+
+    # Retrieve EVSS gi_bill_status data for a user
+    def get_gi_bill_status(user_uuid)
+      user = User.find(user_uuid)
+      service = EVSS::GiBillStatus::Service.new(user)
+      service.get_gi_bill_status
+    rescue => e
+      Rails.logger.error "Failed to retrieve GiBillStatus data: #{e.message}"
+      {}
+    end
+
+    def update_status(submission, status)
+      submission.education_stem_automated_decision.update(automated_decision_state: status)
+    end
+
+    # Makes a list of all submissions that have not been processed and have a status of INIT
+    # Finds most recent submission that has already been processed
+    #
+    # Submissions are marked as processed in EducationForm::CreateDailySpoolFiles
+    #
+    # For each unprocessed submission compare benefit_left and pursuing_teaching_cert values
+    #     to most recent processed submissions
+    # If values are the same set status as PROCESSED
+    # Otherwise check submission data and EVSS data to see if submission can be marked as PROCESSED
+    def check_previous_submissions(submissions, gi_bill_status)
+      unprocessed_submissions = submissions.find_all do |ebc|
+        ebc.processed_at.nil? && ebc.education_stem_automated_decision&.automated_decision_state == INIT
+      end
+      most_recent_processed = submissions.find_all do |ebc|
+        ebc.processed_at.present? && ebc.education_stem_automated_decision&.automated_decision_state != INIT
+      end
+                                         .max_by(&:processed_at)
+
+      processed_form = format_application(most_recent_processed)
+
+      unprocessed_submissions.each do |submission|
+        unprocessed_form = format_application(submission)
+        if unprocessed_form.benefit_left == processed_form.benefit_left &&
+           unprocessed_form.pursuing_teaching_cert == processed_form.pursuing_teaching_cert
```

Comment:
This should also check `is_enrolled_stem`

Reply:
fixed